homeassistant/components/asuswrt/config_flow.py

```diff
@@ -0,0 +1,175 @@
+                    ),
+                    vol.Required(CONF_PORT, default=DEFAULT_SSH_PORT): cv.port,
+                    vol.Required(CONF_MODE, default="router"): vol.In(
+                        {"router": "Router", "ap": "Access Point"}
+                    ),
+                    vol.Required(CONF_REQUIRE_IP, default=True): bool,
+                    vol.Required(CONF_INTERFACE, default=DEFAULT_INTERFACE): str,
+                    vol.Required(CONF_DNSMASQ, default=DEFAULT_DNSMASQ): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    async def _async_check_connection(self, user_input):
+        """Attempt to connect the AsusWrt router."""
+
+        errors = {}
+
+        api = get_api(user_input)
+        try:
+            await api.connection.async_connect()
+            if api.is_connected:
+                if hasattr(api.connection, "disconnect"):
+                    await api.connection.disconnect()
+
+                return self.async_create_entry(
+                    title=self._name,
+                    data=user_input,
+                )
+
+            _LOGGER.error("Error connecting to the AsusWrt router at %s", self._host)
+            errors["base"] = "cannot_connect"
+
+        except OSError:
+            _LOGGER.error("Error connecting to the AsusWrt router at %s", self._host)
+            errors["base"] = "cannot_connect"
+
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception(
+                "Unknown error connecting with AsusWrt router at %s", self._host
+            )
+            errors["base"] = "unknown"
+
+        return self._show_setup_form(user_input, errors)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        errors = {}
+
+        if user_input is None:
+            return self._show_setup_form(user_input, errors)
+
+        host = user_input[CONF_HOST]
+        pwd = user_input.get(CONF_PASSWORD)
+        ssh = user_input.get(CONF_SSH_KEY)
+
+        if not (pwd or ssh):
+            errors["base"] = "pwd_or_ssh"
+        elif ssh:
+            if pwd:
+                errors["base"] = "pwd_and_ssh"
+            else:
+                isfile = await self.hass.async_add_executor_job(_is_file, ssh)
+                if not isfile:
+                    errors["base"] = "ssh_not_file"
+
+        if not errors:
+            ip_address = await self.hass.async_add_executor_job(_get_ip, host)
+            if not ip_address:
+                errors["base"] = "invalid_host"
+
+        if errors:
+            return self._show_setup_form(user_input, errors)
+
+        self._host = host
+        self._name = user_input.get(CONF_NAME, host)
+
+        # Check if already configured
+        await self.async_set_unique_id(ip_address)
```

Comment:
We don't allow setting up addresses as unique id. If there are no valid unique id we can just compare existing entry ip address and abort if the user inputs the same address.

Reply:
I don't have other information available for this. Should I give up multiple entities (see previous comment)?

Reply:
We can store the ip address in the config entry data and iterate all existing entries checking for the ip address in question.

Reply:
Probably do not make sense, I cannot handle multiple ID because I don't have a real UniqueID, so is enough to abort if 1 entry exist

Reply:
Ok.